lib/elixir/lib/io.ex

```diff
@@ -230,9 +238,9 @@ defmodule IO do
   """
   @spec getn(chardata | String.Chars.t, pos_integer) :: chardata | nodata
   @spec getn(device, chardata | String.Chars.t) :: chardata | nodata
-  def getn(prompt, count \\ 1)
+  def getn(device, count \\ 1)
 
-  def getn(prompt, count) when is_integer(count) do
+  def getn(prompt, count) when is_integer(count) and count > 0 do
```

Comment:
ok. I have analyzed the situation and refactoring didn't work.
What we have is the best we can do. I will leave a note in the PR comments